sys/net/network_layer/ng_ipv6/nc/ng_ipv6_nc.c

```diff
@@ -182,4 +182,44 @@ ng_ipv6_nc_t *ng_ipv6_nc_still_reachable(const ng_ipv6_addr_t *ipv6_addr)
     return entry;
 }
 
+int ng_ipv6_nc_translate_mc(void *l2_addr, size_t l2_addr_len,
+                            const ng_ipv6_addr_t *ipv6_addr)
+{
+    uint8_t *l2_u8 = l2_addr;
+
+    if (!ng_ipv6_addr_is_multicast(ipv6_addr)) {
+        return -EADDRNOTAVAIL;
+    }
+
+    switch (l2_addr_len) {
+        case 4:
+            l2_u8[3] = l2_u8[4] = 0xff;
+
+        case 2:     /* just use generic broad cast */
+            l2_u8[1] = 0xff;
+
+        case 1:
+            l2_u8[0] = 0xff;
+            break;
+
+        case 6:
+            l2_u8[0] = l2_u8[1] = 0x33;
+            memcpy(l2_u8 + 2, ipv6_addr->u8 + 12, sizeof(uint32_t));
+            break;
+
+        case 8:
+            l2_u8[0] = l2_u8[1] = 0x33;
+            l2_u8[2] = ipv6_addr->u8[12];
+            l2_u8[3] = 0xff;
+            l2_u8[4] = 0xfe;
+            memcpy(l2_u8 + 4, ipv6_addr->u8 + 13, sizeof(uint32_t));
+            break;
```

Comment:
I could not find anything on EUI-64 translation, so I assumed it to be the same as the EUI-48 scheme with the `ff:fe` insertion.